utils/obsproc/Smap2Ioda.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <iostream>
+#include <netcdf>    // NOLINT (using C API)
+#include <string>
+#include <vector>
+
+#include "eckit/config/LocalConfiguration.h"
+
+#include <Eigen/Dense>    // NOLINT
+
+#include "ioda/Group.h"
+#include "ioda/ObsGroup.h"
+#include "oops/util/DateTime.h"
+
+#include "NetCDFToIodaConverter.h"
+
+namespace gdasapp {
+
+  class Smap2Ioda : public NetCDFToIodaConverter {
+   public:
+    // explicit Smap2Ioda(const eckit::Configuration & fullConfig)
+    // : NetCDFToIodaConverter(fullConfig) {
+    explicit Smap2Ioda(const eckit::Configuration & fullConfig, const eckit::mpi::Comm & comm)
+      : NetCDFToIodaConverter(fullConfig, comm) {
+      variable_ = "Salinity";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by SMAP" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get number of obs
+      int dim0  = ncFile.getDim("phony_dim_0").getSize();
+      int dim1  = ncFile.getDim("phony_dim_1").getSize();
+      int nobs = dim0 * dim1;
+
+      // Set the int metadata names
+      // TODO(AFE) fill this in: intMetadataNames = {"pass", "cycle", "mission"};
+      std::vector<std::string> intMetadataNames = {};
+
+      // Set the float metadata name
+      // TODO(AFE) fill this in:  floatMetadataNames = {"mdt"};
+      std::vector<std::string> floatMetadataNames = {};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
```

Comment:
same comment that I made in @apchoiCMD 's PR:
Change L49 to
```c++
gdasapp::IodaVars iodaVars(nobs, {}, {});
```
and remove L41-47. It's easy enough to add back later.

Reply:
done